src/deps/uws.zig

```diff
@@ -306,6 +306,59 @@ pub fn NewSocketHandler(comptime is_ssl: bool) type {
             );
         }
 
+        /// Get the local address of a socket in binary format.
+        ///
+        /// # Arguments
+        /// - `buf`: A buffer to store the binary address data.
+        /// - `length`: A pointer to an integer representing the length of the buf.
+        ///
+        /// # Returns
+        /// This function returns void, and updated `buf` and `length` inplace.
+        pub fn localAddressBinary(this: ThisSocket, buf: [*]u8, length: *i32) void {
+            return us_socket_local_address(
+                comptime ssl_int,
+                this.socket,
+                buf,
+                length,
+            );
+        }
+
+        /// Get the local address of a socket in text format.
+        ///
+        /// # Arguments
+        /// - `buf`: A buffer to store the text address data.
+        /// - `length`: A pointer to an integer representing the length of the buf.
+        /// - `is_ipv6: A pointer to an boolean representing whether address is IPv6.
+        ///
+        /// # Returns
+        /// This function returns void, and updated `buf` and `length` inplace.
+        pub fn localAddressText(this: ThisSocket, buf: [*]u8, length: *i32, is_ipv6: *bool) void {
```

Comment:
i wish this code was 'safer'. i think the handling of memory is ok here after reading it very carefully.

maybe it should return a slice instead of mutating a length pointer? this way it would be harder for the caller to access the undefined memory in `buf`.

Reply:
Now it returns a slice on success.
```zig
        pub fn localAddressText(this: ThisSocket, buf: []u8, is_ipv6: *bool) ?[]const u8 {

```